### PR TITLE
Starts splitting config helpers from the config package

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

As we continue to migrate the Agent code to components the many helpers in pkg/config need to be moved somewhere else. The goal is to have a simple API for the Config component while offering helpers on top of it.

The goal is to have helpers that can take either a "pkg/config:Config" or a "comp/core/config:Component". This will facilitate the migration to components.

For now the helpers are moved to pkg/config/utils. This is temporary as we brainstorm the best place for them. Starting to split them from the main config package is still useful and unblock the development of some components.

### Describe how to test/QA your changes

Test that the agent still uses the right URL to send data.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/settings/runtime_settings_test.go [hunk 1]
```diff
@@ -105,7 +98,7 @@ func TestLogLevel(t *testing.T) {
 
 func TestProfiling(t *testing.T) {
 	cleanRuntimeSetting()
-	setupConf()
+	config.SetupConf()
```
Comment (gh123man) [inline pkg/config/settings/runtime_settings_test.go:101]: In the near future, I think we can replace this function 1:1 with `cfg := fxutil.Test[config.Component](t, config.MockModule)` 😄

**Review by gh123man — APPROVED** [review]

File: pkg/config/settings/runtime_settings_test.go (hunk 1 above)
Comment (gh123man) [reply pkg/config/settings/runtime_settings_test.go:101]: for sure. definitely a task for later 👍

**Review by purple4reina — APPROVED** [review]

**Review by sblumenthal — APPROVED** [review]
lgtm for container integration file changes

**Review by ameske — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by just-chillin — APPROVED** [review]
Looks good! Just left one suggestion. We're already using the config component everywhere in `pkg/process`, so make sure you try to use it when you can please 🙏🏻

**Review by sarah-witt — APPROVED** [review]

File: pkg/config/utils/endpoints.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		},
+	}
+
+	additionalEndpoints := c.GetStringMapStringSlice("additional_endpoints")
+	return mergeAdditionalEndpoints(keysPerDomain, additionalEndpoints)
+}
+
+// GetMainEndpoint returns the main DD URL defined in the config, based on `site` and the prefix, or ddURLKey
+func GetMainEndpoint(c config.ConfigReader, prefix string, ddURLKey string) string {
+	return getEndpoint(c, prefix, ddURLKey)
```
Comment (ogaca-dd) [inline pkg/config/utils/endpoints.go:115]: As `GetMainEndpoint` is exactly the same as `getEndpoint `, `getEndpoint` can be removed.

**Review by ogaca-dd — APPROVED** [review]
LGTM.
Note: I assumed the tests were moved and I didn't review them

**Pull request merged**